src/libraries/System.Net.Security/ref/System.Net.Security.cs

```diff
@@ -33,7 +33,7 @@ public enum EncryptionPolicy
         AllowNoEncryption = 1,
         NoEncryption = 2,
     }
-    public delegate System.Security.Cryptography.X509Certificates.X509Certificate LocalCertificateSelectionCallback(object sender, string targetHost, System.Security.Cryptography.X509Certificates.X509CertificateCollection localCertificates, System.Security.Cryptography.X509Certificates.X509Certificate remoteCertificate, string[] acceptableIssuers);
+    public delegate System.Security.Cryptography.X509Certificates.X509Certificate LocalCertificateSelectionCallback(object sender, string targetHost, System.Security.Cryptography.X509Certificates.X509CertificateCollection localCertificates, System.Security.Cryptography.X509Certificates.X509Certificate? remoteCertificate, string[] acceptableIssuers);
```

Comment:
I presume `sender` should be declared nullable here?

Reply:
Do we ever invoke this will null as the sender?  Are there any situations where any code you've found passes null?  Is there code that creates these delegates not defensively checking that sender is null?

Reply:
See https://source.dot.net/#System.Net.Security/System/Net/Security/SecureChannel.cs,655
Here `string.Empty` is used as `sender`.

See https://source.dot.net/#System.Net.Security/System/Net/Security/SecureChannel.cs,360
Here TargetHost is used as `sender`. Default for the TargetHost is string.Empty https://source.dot.net/#System.Net.Security/System/Net/Security/SslAuthenticationOptions.cs,47

Reply:
Non-nullable it is then.